plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/Details.tsx

```diff
@@ -57,12 +57,12 @@ const DatasetDetails: React.FC<Props> = ({
       )}
     </>
   );
-  return dataset ? (
+  return dataset && ("dataID" in dataset || dataset.desc) ? (
     <DetailsComponent
       dataset={dataset}
-      addDisabled={addDisabled(dataset.dataID)}
+      addDisabled={"dataID" in dataset && addDisabled(dataset.dataID)}
```

Comment:
Hm? Doesn't this mean that if there is no dataID it will never be disabled??

Reply:
@KaWaite , I'm hiding this section inside the `DetailsComponent ` so this parameter doesn't matter for items with no `dataID`.

Reply:
Then you don't need this change, right? Its the same as in a conditional like this `if(dataset && dataset.components)` you don't need to do `dataset?.components` beacuse you have already ruled out an undefined dataset, right?

Reply:
@KaWaite , actually this is needed, because `addDisabled ` function needs `dataID` to work, and `dataset` could be a `DataCatalogGroup ` so it will give a typing error even if you do something like you mentioned:
`Property 'dataID' does not exist on type 'DataCatalogItem | DataCatalogGroup'.`

Reply:
Ah I see now. In the return's conditional we allow through even if dataset has no dataId, but has a desc. In that case, we won't have dataID. Gotcha